libraries/chain/include/steem/chain/smt_objects/smt_token_object.hpp

```diff
@@ -44,30 +44,34 @@ class smt_token_object : public object< smt_token_object_type, smt_token_object
       c( *this );
    }
 
-   price    one_vesting_to_one_liquid() const
+   price initial_vesting_share_price() const
    {
-      int64_t one_smt = std::pow(10, liquid_symbol.decimals());
-      return price ( asset( one_smt, liquid_symbol.get_paired_symbol() ), asset( one_smt, liquid_symbol ) );
-      // ^ On the assumption that liquid and vesting SMT have the same precision. See issue 2212
+      int64_t one_smt = std::pow( 10, liquid_symbol.decimals() );
+      return
+         price(
+            asset( one_smt * SMT_INITIAL_VESTING_PER_UNIT, liquid_symbol.get_paired_symbol() ),
+            asset( one_smt, liquid_symbol )
+         );
    }
 
-   price    get_vesting_share_price() const
+   price get_vesting_share_price() const
    {
-      if ( total_vesting_fund_smt == 0 || total_vesting_shares == 0 )
-         return one_vesting_to_one_liquid();
-         // ^ In original method of globa_property_object it was one liquid to one vesting which seems to be a bug.
+      if ( total_vesting_fund_ballast == 0 || total_vesting_shares_ballast == 0 )
+         return initial_vesting_share_price();
 
-      return price( asset( total_vesting_shares, liquid_symbol.get_paired_symbol() ), asset( total_vesting_fund_smt, liquid_symbol ) );
+      share_type effective_total_vesting_shares = total_vesting_shares_ballast + total_vesting_shares;
+      share_type effective_total_vesting_fund   = total_vesting_fund_ballast + total_vesting_fund_smt;
+
+      return price( asset( effective_total_vesting_shares, liquid_symbol.get_paired_symbol() ), asset( effective_total_vesting_fund, liquid_symbol ) );
    }
 
-   price    get_reward_vesting_share_price() const
+   price get_reward_vesting_share_price() const
```

Comment:
The formula for reward vesting price for STEEM is:

```c++
return price( total_vesting_shares + pending_rewarded_vesting_shares,
   total_vesting_fund_steem + pending_rewarded_vesting_steem );
```

For an SMT it should be:

``` c++
return price( total_vesting_shares_ballast + total_vesting_shares + reward_vesting_shares,
   total_vesting_fund_ballast + total_vesting_fund_smt + reward_vesting_smt );
```